integrated_channels/integrated_channel/management/commands/transmit_learner_data.py

```diff
@@ -0,0 +1,96 @@
+"""
+Transmits consenting enterprise learner data to the Enterprise Integrated Channel SAP SuccessFactors.
+"""
+from __future__ import absolute_import, unicode_literals
+
+from django.core.management.base import BaseCommand, CommandError
+
+from enterprise.models import EnterpriseCustomer
+from integrated_channels.sap_success_factors.models import SAPSuccessFactorsEnterpriseCustomerConfiguration
+
+try:
+    from djcelery import celery
+except ImportError:
+    celery = None
+
+
+class Command(BaseCommand):
+    """
+    Management command which transmits learner course completion data to the IntegratedChannel(s) configured for the
+    given EnterpriseCustomer.
+
+    Collect the enterprise learner data for enrollments with data sharing consent, and transmit each to the
+    EnterpriseCustomer's configured IntegratedChannel(s).
+    """
+    help = """
+    Transmit enterprise learner course completion data for the given enterprise customer.
+    """
+    integrated_channel_choices = {
+        'sap': SAPSuccessFactorsEnterpriseCustomerConfiguration,
+    }
+
+    def add_arguments(self, parser):
+        parser.add_argument(
+            '--enterprise_customer',
+            dest='enterprise_customer',
+            default=None,
+            metavar='ENTERPRISE_CUSTOMER_UUID',
+            help='Transmit data for this EnterpriseCustomer to its configured integration channel(s).',
+        )
+        parser.add_argument(
+            '--channel',
+            dest='channel',
+            default=None,
+            metavar='INTEGRATED_CHANNEL',
+            help='Transmit data to this INTEGRATED_CHANNEL.  '
+            'Omit option to transmit to all configured, active integration channels.',
+            choices=self.integrated_channel_choices.keys(),
+        )
+
+    def handle(self, *args, **options):
+        """
+        Collect the learner data for the given EnterpriseCustomer, and transmit it to its active integration channels.
+        """
+        # Ensure there's a valid enterprise customer
+        enterprise_customer_uuid = options['enterprise_customer']
+        if enterprise_customer_uuid:
+            try:
+                enterprise_customer = EnterpriseCustomer.active_customers.get(uuid=enterprise_customer_uuid)
+            except EnterpriseCustomer.DoesNotExist:
+                raise CommandError('Enterprise customer {} not found, or not active'.format(enterprise_customer_uuid))
+        else:
+            enterprise_customer = None
+
+        channel_code = options['channel']
+        if channel_code is not None:
+            if channel_code not in self.integrated_channel_choices:
+                raise CommandError('Invalid integrated channel: {}'.format(channel_code))
+            plugin_classes = [self.integrated_channel_choices.get(channel_code)]
+        else:
+            # Otherwise, use all the configured channel types
```

Comment:
imho, this comment would read better if we instead explain what is going on overall, just prior to the `if` statement.